### ISSUE TITLE
Update and revise the GeoNode Ansible deployment

### ISSUE THREAD

**Issue by gannebamm** [GeoNode/geonode#6705]
**Is your feature request related to a problem? Please describe.**
The current ansible-playbook is old and not maintained anymore. It should get revised by an experience DevOp as part of a fork which than can lead to a GNIP.

**Describe the solution you'd like**
The ansible-playbook should be more granular and enable eg. updating GeoServer if there are any security fixes for it.

**Describe alternatives you've considered**
There is already a working docker-composition which can be used for DevOps. Since docker does not provide enough control for some DevOps the ansible role should be an alternative.

**Additional context**

**Comment by gannebamm** [comment]
https://github.com/malnajdi will work on this

**Comment by t-book** [comment]
... if the ansible role is generic enough it could even be used to for example update geoserver inside a docker container. just my two cents.

**Comment by gannebamm** [comment]
@malnajdi had some very nice ideas about the 'genericness' and updating GeoServer per Ansible was one of them. If the docker composition will also be covered is TBD (I do not think so), but maybe that could be an addition done by others.
If you have anything to show or discuss @malnajdi do not hesitate to link it here.

**Comment by malnajdi** [comment]
Great, so my initial ideas and we can go on about this and discuss it more later but here it is:

1. We need to generalize the roles or maybe even use an roles made already by the community.
2. We need to ensure that the roles are designed in a way where it's easy to add support for different Linux flavor mainly now Debian and Ubuntu. 
3. I like to think of the components we have as (1. Something more closer to the Dev) (2. something more closer to the Ops). 

With the things above in mind GeoNode Template should have a deploy folder where inside that folder you would be able to 
deploy that django application to a target server and the target server is managed by the Ops team. 

The other components like postgres and geoserver we can think of them as services provided to the developer as [ip or domain]
where the developer can use these [ip or domain] in their GeoNode template. 

The deploy folder will also be used to do a lot of things:
- prepare a docker image docker.yml
- prepare a postgres database for that GeoNode Template database.yml
- Deploy the django application application.yml
... and think of the possibilities that we can do.

The current roles has a lot of old structure and i guess as @gannebamm has pointed out it's out dated.

To summarize:
This role can be changed to: 

postgres [ which has an option to enable/install postgis extension & option to install it via docker ?? maybe ??]
geoserver [ which has an option to enable/install extra extension & option to install it via docker ?? maybe ??]
apache [ which can be used by geonode and geoserver if needed]
nginx [ which can be used by geonode and geoserver if needed]
uwsgi [ which will be used by django-role ]
celery [ which will be used by django-role if needed]
django-role [deploy a django application which has a dependency of uwsgi and [nginx or apache] & option to install it via docker ?? maybe ??]

As always please let me know if you think something is not correct or need to be changed etc..

**Comment by gannebamm** [comment]
I have read on gitter (https://gitter.im/GeoNode/general?at=603de94cd1aee44e2dbb4a8c) the ansible deployment is currently used on your site. Do you think there is the possibility to upstream some of it?

**Comment by malnajdi** [comment]
> I have read on gitter (https://gitter.im/GeoNode/general?at=603de94cd1aee44e2dbb4a8c) the ansible deployment is currently used on your site. Do you think there is the possibility to upstream some of it?

What i did is to automate the docker deployment instead of using docker-compose, i rely on ansible. so it's not the core deployment without docker.

**Comment by giohappy** [comment]
Ansible has been restored for geonode-project. 
Can we close this?

**Comment by gannebamm** [comment]
I am ok with closing this
@malnajdi ?

**Comment by malnajdi** [comment]
go ahead
